scanner/go.go

```diff
@@ -1,19 +1,51 @@
 package scanner
 
+import (
+	"fmt"
+	"golang.org/x/mod/modfile"
+	"os"
+)
+
 func configureGo(sourceDir string, config *ScannerConfig) (*SourceInfo, error) {
-	if !checksPass(sourceDir, fileExists("go.mod", "Gopkg.lock")) {
+	if !checksPass(sourceDir, fileExists("go.mod")) {
 		return nil, nil
 	}
 
+	gomod, parseErr := parseModfile()
+
+	version := "1"
+	if parseErr == nil && len(gomod.Go.Version) > 0 {
```

Comment:
nitpick: an empty string is usually compared as `gomod.Go.Version != ""` rather than on its length 🤷

Reply:
on another note, parseErr is swallowed, no idea how other scanners handles this but not showing a warning with the error may lead to broken image builds  as go.mod isn't valid anyway.